pkg/controllers/dynakube/injection/reconciler.go

```diff
@@ -132,9 +135,11 @@ func (r *reconciler) setupOneAgentInjection(ctx context.Context) error {
 		return err
 	}
 
-	err = r.pmcSecretreconciler.Reconcile(ctx)
-	if err != nil {
-		return err
+	if !r.dk.FeatureDownloadViaJob() || r.dk.OneAgent().IsCSIAvailable() {
```

Comment:
```suggestion
	if !r.dk.FeatureBootstrapperInjection() {
```

you are using the `r.dk.FeatureDownloadViaJob() || r.dk.OneAgent().IsCSIAvailable()` check often, you should abstract it

Reply:
https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/349bd42a7afbecd1850b936cfa50bc74e8f49892